### PR TITLE
Feature/kx 54 add a detail page for each document

### PR DESCRIPTION
**Pull request by GiuseppeArbore**

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+import { NotFound } from "./components/NotFound";
 import { Route, Routes } from 'react-router-dom';
 
 export default function App() {
 
   return (
     <Routes>
       <Route path="/" element={<Console />} />
+      <Route path="/documents/:id" element={<Document />} />
+      <Route path="/page-not-found" element={<NotFound />} />
```
Comment (lbordino) [inline client/src/App.tsx:12]: probably it's better to use path="*"  also style it a bit

File: client/src/components/Document.tsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+            setType(document.type);
+            setLanguage(document.language || "");
+            setPages(document.pages || "");
+            setDescription(document.description || "");
+        }).catch((error) => {
+            console.error("PIZZA Failed to fetch document:", error);
+            //window.location.href = "/page-not-found";
+        });
+    } );
+
```
Comment (lbordino) [inline client/src/components/Document.tsx:48]: use async await paradigm

File: client/src/components/Document.tsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                    <div className='hidden lg:flex flex-col items-start col space-y-2 w-1/2'>
+                        <i className="text-md font-light text-tremor-content-strong dark:text-dark-tremor-content-strong">Description:</i>
+                        <i className='text-sm font-light text-tremor-content-strong dark:text-dark-tremor-content-strong'> {description}  </i>
+                    </div>
+                    <i onClick={() => setShare(true)}><RiShareLine className="text-2xl text-tremor-content-strong dark:text-dark-tremor-content-strong" /></i>
+                    <Dialog open={share} onClose={() => setShare(false)} static={true}>
+                        <DialogPanel>
+                            <h3 className="text-lg font-semibold text-tremor-content-strong dark:text-dark-tremor-content-strong">Share "{title}"</h3>
+                            <p className="mt-2 leading-6 text-tremor-default text-tremor-content dark:text-dark-tremor-content">
+                                Share the link of the document with your friends and colleagues.
```
Comment (lbordino) [inline client/src/components/Document.tsx:103]: remove friends

File: client/src/components/Document.tsx [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+                                        className="w-full p-2 border border-tremor-border rounded-md"
+                                        value={`http://localhost:1420/documents/${id}`}
+                                        readOnly
+                                    />
+                                </div>
+
+                                <Button
+                                    className="mt-4 w-1/6 flex flex-col items-center justify-between"
+                                    onClick={() => {
+                                        navigator.clipboard.writeText(`http://localhost:1420/documents/${id}`);
```
Comment (lbordino) [inline client/src/components/Document.tsx:118]: don't use localhost but window.location

**Review by lbordino — CHANGES_REQUESTED** [review]
- Add a couple of tests
- Style the 404 page 
- Add a title or something above the card stating Page Title
- When asking for a unknown doc it should redirect to 404 page
- If a field e.g. pages is missing either remove it or insert a placeholder like "Unknown"

File: client/src/App.tsx (hunk 1 above)
Comment (GiuseppeArbore) [reply client/src/App.tsx:12]: you're right

File: client/src/components/Document.tsx (hunk 3 above)
Comment (GiuseppeArbore) [reply client/src/components/Document.tsx:103]: So, also colleagues... it's not specified...

File: client/src/components/NotFound.tsx
[the remark this answers is not in the document]
Comment (GiuseppeArbore) [reply client/src/components/NotFound.tsx:1]: It's temporary page but i'll change it

**Review by lbordino — APPROVED** [review]

**Pull request merged**